examples/data-jpa-repository/src/test/kotlin/cnt/jpa/domain/MovieRepositoryWithH2Test.kt

```diff
@@ -0,0 +1,54 @@
+package cnt.jpa.domain
+
+import util.H2TestProfile
+import util.TransactionalQuarkusTest
+import io.quarkus.test.common.QuarkusTestResource
+import io.quarkus.test.h2.H2DatabaseTestResource
+import io.quarkus.test.junit.TestProfile
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import javax.inject.Inject
+
+@TransactionalQuarkusTest
+@TestProfile(H2TestProfile::class)
+@QuarkusTestResource(H2DatabaseTestResource::class)
+class MovieRepositoryWithH2Test {
+
+    @Inject
+    lateinit var movieRepository: MovieRepository
+
+    @BeforeEach
+    fun clearDatabase() {
+        movieRepository.deleteAll()
```

Comment:
Question: if the test itself is transactional (rollback at the end of each test like in Spring?) is this cleanup strictly necessary?

Reply:
the rollback does not happen automatically

Reply:
I understand it like this:
with the annotation `@TransactionalQuarkusTest` each method is declared as transactional which means that after each test method a full commit and flush to database happens.
without the transactional annotation, i would only test the entity manager behavior, but would not see if the insert really works (for complex inserts)

Reply:
Good to know